sdk/src/main/java/io/customer/sdk/extensions/RetrofitExtensions.kt

```diff
@@ -14,11 +14,15 @@ private fun <T> Response<T>.bodyOrThrow(): T {
 
 private fun <T> Response<T>.toException() = HttpException(this)
 
-fun <T> Response<T>.toResultUnit(): Result<Unit> = toResult { }
+fun <T> Response<T>.toCompatibleResultUnit(): Result<Unit> = toCompatibleResult { }
 
-fun <T> Response<T>.toResult(): Result<T> = toResult { it }
+fun <T> Response<T>.toResultUnit(): kotlin.Result<Unit> = toResult { }
 
-fun <T, E> Response<T>.toResult(mapper: (T) -> E): Result<E> {
+fun <T> Response<T>.toCompatibleResult(): Result<T> = toCompatibleResult { it }
+
+fun <T> Response<T>.toResult(): kotlin.Result<T> = toResult { it }
```

Comment:
From my understanding of the SDK, the custom `Result` class only exists so that the SDK has backwards compatibility with Java. 

Assuming that is the reason why we have a custom `Result` class, this is why I added `kotlin.Result`. 

When the background queue is fully implemented, the public facing functions of the SDK are all synchronous. No need to return any type of `Result` to the user. At that time, `Result` will be only used internally in the SDK and not public facing. Because it is only internal, the SDK can get away with only using the `kotlin.Result` type. At that time, I was hoping we can delete the custom `Result` type. 

The idea is to make our lives easier with maintaining the SDK. If we only need 1 type of `Result`, that's easier to manage then having to convert back and forth between the two in the code base.

Reply:
Sure thing, as long as we are sticking with one either `Result` or `kotlin.Result`. 

> No need to return any type of Result

I do believe this would be the case for most of the things, but I expect some methods to be done asynchronously, or for us to provide a callback for some methods, and we need to have some room for this to tackle in the future, otherwise if we follow this approach aggressively we might have to make a lot of changes in future to tackle other cases.

Reply:
I agree that this scenario may happen in the future where we need to return a `Result` in a public SDK function. 

With the current feature set of the SDK, luckily, we don't have that scenario so we have some time to think about what to do. 

> we might have to make a lot of changes in future to tackle other cases.
If we use `kotlin.Result` internally in the SDK code and result a custom `Result` type right at the point of returning a result from a public function:


 I don't see us needing to make a lot of changes. I am assuming you mean we would need to make a lot of changes if we wanted to change `kotlin.Result` use internally with something else.